### PR TITLE
Make doc-fmt iterate until all is fixed

### PR DESCRIPTION
**Pull request by JAremko**
Now it should fix all the errors before exiting.

Also added an example to test.org of an error that can be fixed only with multiple iterations and fixed bash warning.

### CODE REVIEW

File: core/tools/doc-fmt/run.bash [hunk 1]
```diff
@@ -12,25 +12,37 @@ if hash gsed 2>/dev/null; then
...
-   find $place -name "*.org" -type f -exec $seder -i 's/[ \t]*$//' {} \;
-   #Remove #+HTML_HEAD_EXTRA: ... readtheorg.css" />
-   find $place -name "*.org" -type f -exec $seder -i '/#+HTML_HEAD_EXTRA.*readtheorg.css.*/d' {} \;
-   #Replace multiply empty lines with a single empty line
-   find $place -name "*.org" -type f -exec $seder -i '/^$/N;/^\n$/D' {} \;
-   #Replace :TOC_4_org: with :TOC_4_gh:
-   find $place -name "*.org" -type f -exec $seder -i 's/:TOC_4_org:/:TOC_4_gh:/' {} \;
-   #apply toc-org
-   find $place -name "*.org" -type f -exec emacs -batch -l ./core/tools/doc-fmt/fmt.el '{}' -f apply-all \;
+   is_done=1
```
Comment (JAremko) [inline core/tools/doc-fmt/run.bash]: Unused variable.

**Comment by JAremko** [comment]
Travis  why you no... :cry:

**Comment by JAremko** [comment]
Travis, you did it! :clap:

**Comment by syl20bnr** [comment]
Thank you again :-)
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**